### PR TITLE
Fixes #110: Prevent UnicodeEncodeError

### LINKED ISSUE #110: Bug: UnicodeEncodeError when redirect stdout to a file
## Description
I'm trying to use Halo in a CLI that can be called from a terminal directly or from a CI, when I try to redirect the **stdout** to a file I have this exception.

Example:

```python
def my_cli_task:
    spinner = halo.Halo(enabled=True, spinner='dots')
    spinner.start(text="Starting")
    ...
    
    ...
    spinner.succeed('Complete')
    spinner.stop()
```
It works perfectly in the terminal but when I try to redirect the stdout to a file, it raise an exception.
e.g.:
```bash
$ ./my_example_cli_task.py > cli.log
```

### System settings

- Operating System: Ubuntu 18.04.1 LTS
- Terminal in use: NA
- Python version: Python 2.7.15rc1
- Halo version: 0.0.21
- `pip freeze` output:

### Error

```bash
 Traceback (most recent call last):
  File "test.py", line 7, in <module>
    spin.start(text="Starting")
  File "/home/agustin/git/halo/halo/halo.py", line 423, in start
    self._render_frame()
  File "/home/agustin/git/halo/halo/halo.py", line 341, in _render_frame
    self._stream.write(output)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 166, in write_and_convert
    self.write_plain_text(text, cursor, start)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "/usr/local/lib/python2.7/dist-packages/colorama/ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
UnicodeEncodeError: 'ascii' codec can't encode character u'\u280b' in position 0: ordinal not in range(128)
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/agustin/git/halo/halo/halo.py", line 81, in clean_up
    self.stop()
  File "/home/agustin/git/halo/halo/halo.py", line 440, in stop
    self._spinner_thread.join()
  File "/usr/lib/python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
RuntimeError: cannot join thread before it is started
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/agustin/git/halo/halo/halo.py", line 81, in clean_up
    self.stop()
  File "/home/agustin/git/halo/halo/halo.py", line 440, in stop
    self._spinner_thread.join()
  File "/usr/lib/python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
RuntimeError: cannot join thread before it is started
```
### Expected behaviour

Print the my CLI log to a file
## Steps to recreate

Redirect the stdout to file
## People to notify

@alarreine

### PR DESCRIPTION
**Pull request by alarreine**
## Description of new feature, or changes

Fix #110 
## Checklist

- [x] Your branch is up-to-date with the base branch
- [x] You've included at least one test if this is a new feature
- [x] All tests are passing

## Related Issues and Discussions

Resolves #110
## People to notify

@manrajgrover

### CODE REVIEW

**Comment by manrajgrover** [comment]
@alarreine Thanks for raising the issue and for the fix! 😄

You would need to add tests for this bug fix before this can be merged. Let me know if you need any assistance.

**Comment by alarreine** [comment]
Hi @manrajgrover 
I added the test in commit a26b58d

File: halo/_utils.py [hunk 1]
```diff
@@ -115,6 +115,25 @@ def decode_utf_8_text(text):
         return text
 
 
+def encode_utf_8_text(text):
+    """Decode the text from utf-8 format
```
Comment (manrajgrover) [inline halo/_utils.py]: This should be:

```suggestion
    """Encodes the text to utf-8 format
```

File: halo/_utils.py [hunk 2]
```diff
@@ -115,6 +115,25 @@ def decode_utf_8_text(text):
...
+
+     Parameters
```
Comment (manrajgrover) [inline halo/_utils.py]: Remove extra spacing here:

```suggestion
    Parameters
```

**Pull request merged**